### DISCUSSION TITLE
change port from 3000 to 443

### DISCUSSION THREAD

**Discussion by jayvgithub** [KiraLT/torrent-stream-server#217] [Q&A] [answered]
want to run this on ssl port 443 how can i change port from 3000 to 443

**Comment by KiraLT** [comment, marked as answer]
To change the port you can provide `config.json` file or specify the `PORT=443` env variable. You can find more information in [documentation](https://github.com/KiraLT/torrent-stream-server/wiki/Configuration)

If you are using docker, check docker [documentation](https://github.com/KiraLT/torrent-stream-server/wiki/Setup#custom-port).

However, simply by changing the port SSL won't activate. At the moment server **doesn't support SSL**. The easiest way to have SSL is to place a proxy in front of the server with SSL support. Or you can just use [cloudflare](https://www.cloudflare.com/) (use DNS to set up HTTPS to HTTP proxy). Find more in the [docs](https://github.com/KiraLT/torrent-stream-server/wiki/Setup#ssl--custom-domain).

**Comment by jayvgithub** [comment]
is there any option i can streaming with out downloading to local server and stream from seeders are you available on telegram

**Comment by omgbox** [comment]
> is there any option i can streaming with out downloading to local server and stream from seeders are you available on telegram

at this moment that is not possible, i think you looking for jstorrent addon for chrome browser, this will directly use the chrome browser to download your torrent and not use any web servers. 

torrent-stream-server requires a web server to function . it needs to save data into folder and re-send back to you viewer in browser. so you see .

hopefully soon every torrent client will add the WebRTC support , this will enable you to do just that with just a modern browser. have a look at this instant.io this is a pure javascript webrtc torrent client. it is not able right now to talk to normal UDP peers like desktop torrent clients. only web browsers to web browsers communications.
